Performance Troubleshooting Notebook

jcpkanuri opened the pull request:
Database Performance Troubleshooting notebook serves as a base to build upon to aid Database Administrators and Developers to help with identifying common database performance-related issues.
This version supports ad-hoc execution of the Notebook and the next version can be scheduled.

kesmit13 reviewed (changes requested):
You'll need to run the pre-commit checks on the files before they can be published. See https://github.com/singlestore-labs/spaces-notebooks/blob/master/CONTRIBUTING.md for more information. 

As far as notebook changes:

1) Is the CSV files located remotely so that it can be updated separately from the notebook? They could simply be embedded in the notebook as Python code, or write a local CSV file using a `%%writefile` cell which would keep the notebook self-contained.

2) The `templates` link in the introduction seems to point to a folder that is not public.

3) You don't need the `my_db_conn_url` variable. There is a `connection_url` variable in the notebook scope. To get the database name you can just do `connection_url.split('/')[-1]`. However, that only works if a database is selected in the notebook. There may not be a database path specified in the URL. The most correct way would be to use `urllib.parse.urlparse` and get the path component of the result.

4) I would remove the `if __name__` line. Since this is in a notebook, it can't be loaded as a library, so this just adds an extra indent that isn't needed. It might be a good idea to split out the imports, the variable initialization, the function definitions, and the script code into separate cells to make it more "notebook-like". This would allow you to add some commentary on each section as well.

5) I know you have just one big block of code so you can do error handling, but can you add some comments to sections of the code explaining what's going on?

6) You don't need a Management API key to access that API. In the notebook context, it will use the current JWT to authenticate. If you do want to use an API key, you can set it in the environment under the `SINGLESTOREDB_MANAGEMENT_TOKEN` name. This could be pulled from a secret as well so no prompt is necessary.

jcpkanuri commented:
Hi Kevin,
Please see my responses inline to your questions.

*Jai Kanuri*
Sr. Professional Services Architect
m. 512-784-8838
[image: SingleStore. The Database of Now.]
<https://www.singlestore.com/managed-service-trial/?utm_source=emailsig>


On Fri, Mar 1, 2024 at 9:19 AM Kevin D Smith ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> You'll need to run the pre-commit checks on the files before they can be
> published. See
> https://github.com/singlestore-labs/spaces-notebooks/blob/master/CONTRIBUTING.md
> for more information.
>
> As far as notebook changes:
>
>    1.
>
>    Is the CSV files located remotely so that it can be updated separately
>    from the notebook? They could simply be embedded in the notebook as Python
>    code, or write a local CSV file using a %%writefile cell which would
>    keep the notebook self-contained. -- The CSV files were hosted on S3.
>    I switched them over to Github
>    2.
>
>    The templates link in the introduction seems to point to a folder that
>    is not public. -- I moved the folders to Github and they are
>    accessible now
>    3.
>
>    You don't need the my_db_conn_url variable. There is a connection_url
>    variable in the notebook scope. To get the database name you can just do
>    connection_url.split('/')[-1]. However, that only works if a database
>    is selected in the notebook. There may not be a database path specified in
>    the URL. The most correct way would be to use urllib.parse.urlparse
>    and get the path component of the result. -- I fixed this issue
>    4.
>
>    I would remove the if __name__ line. Since this is in a notebook, it
>    can't be loaded as a library, so this just adds an extra indent that isn't
>    needed. It might be a good idea to split out the imports, the variable
>    initialization, the function definitions, and the script code into separate
>    cells to make it more "notebook-like". This would allow you to add some
>    commentary on each section as well. -- I fixed this issue
>    5.
>
>    I know you have just one big block of code so you can do error
>    handling, but can you add some comments to sections of the code explaining
>    what's going on? -- I added the comments to sections
>    6.
>
>    You don't need a Management API key to access that API. In the
>    notebook context, it will use the current JWT to authenticate. If you do
>    want to use an API key, you can set it in the environment under the
>    SINGLESTOREDB_MANAGEMENT_TOKEN name. This could be pulled from a
>    secret as well so no prompt is necessary. -- I updated the code to use
>    JWT to authenticate
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/singlestore-labs/spaces-notebooks/pull/65#pullrequestreview-1911405092>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALY7KZ2GFHTEKG25FYNLE6TYWCL7ZAVCNFSM6AAAAABD77YG2KVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMJRGQYDKMBZGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
